### ISSUE TITLE
installation , process ?

### ISSUE THREAD

**Issue by F1SMV** [stdevPavelmc/FAASGS#12]
Bonjour

I followed the different steps in order to install your distribution but nothing starts at this moment..I installed, predict, wxotimg but on arrival there is nothing in var / www .. elsewhere the instruction in FAASGS is ./sat.sh (or sudo ./sat.sh) and not sats.sh as indicated in the paragraph "real install"
how can we be sure that the installation is ok? indeed when I type the ip address, I come across the page of the nginx server
thank for your infos
Eric

**Comment by stdevPavelmc** [comment]
Hi Eric

Please update the install and please follow the steps one by one.

I think you are missing the steps in the `make config`, `make install`, `make persistent`

Write back if any trouble

**Comment by F1SMV** [comment]
Thank Pavel, i will do..this we.

**Comment by stdevPavelmc** [comment]
Any progress?

**Comment by F1SMV** [comment]
gm
i try sudo git pull or git pull in /FAASGS but raspi ask me to valid fusion on sat.sh and sat_data/user.conf how to intervene ?
thank in advance

**Comment by stdevPavelmc** [comment]
Just run this command in the FAASGS folder to reset changes and then the git pull:

```sh
git reset --hard
git pull
```

**Comment by F1SMV** [comment]
thank, its ok for that, but when i want to see my homepage its a nginx page "welcome" which appears..

**Comment by stdevPavelmc** [comment]
You need to run this options to update the install to the new settings

- sudo make clean
- sudo make install

Configure the /etc/sat_data/user.conf again (see README.md)

Finally run:

- sudo make permanent

**Comment by F1SMV** [comment]
yep but...always ngninx frontpage when i type raspi ip adress...:(

**Comment by takagiwa** [comment]
Hello,

I think "index.php" not added or on later than "index.html" in "index" parameter somewere in nginx configuration file.

In my environment (Raspberry Pi OS), I have installed nginx, php and php-fpm.
And modify part of /etc/nginx/sites-available/default as below

> \# Add index.php to the list if you are using PHP
> index index.php index.html index.htm index.nginx-debian.html;

> location ~ \.php$ {
>     include snippets/fastcgi-php.conf;
> 
>     \# With php-fpm (or other unix sockets):
>     fastcgi_pass unix:/run/php/php7.3-fpm.sock;
> \#   \# With php-cgi (or other tcp sockets):
> \#   fastcgi_pass 127.0.0.1:9000;
> }

and run
> $ sudo /etc/init.d/php7.3-fpm restart
> $ sudo /etc/init.d/nginx restart

I hope this will be of some help.

**Comment by F1SMV** [comment]
thank for info but...after sudo /etc/init.d/ngninx.restart i had this message 👍 
sudo /etc/init.d/nginx restart
[....] Restarting nginx (via systemctl): nginx.serviceJob for nginx.service failed because the control process exited with error code.
See "systemctl status nginx.service" and "journalctl -xe" for details.
 failed!

i send cmd : systemctl status nginx.service

● nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Thu 2021-02-11 13:37:21 CET; 1min 25s ago
     Docs: man:nginx(8)
  Process: 1883 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)

févr. 11 13:37:21 raspberrypi systemd[1]: Starting A high performance web server and a reverse proxy server...
févr. 11 13:37:21 raspberrypi nginx[1883]: nginx: [emerg] unexpected end of file, expecting "}" in /etc/nginx/sites-enabled/defaul
févr. 11 13:37:21 raspberrypi nginx[1883]: nginx: configuration file /etc/nginx/nginx.conf test failed
févr. 11 13:37:21 raspberrypi systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
févr. 11 13:37:21 raspberrypi systemd[1]: nginx.service: Failed with result 'exit-code'.
févr. 11 13:37:21 raspberrypi systemd[1]: Failed to start A high performance web server and a reverse proxy server.

can you help me to fix it ? thank

**Comment by takagiwa** [comment]
Hello,

It looks there are syntax error on nginx configuration file.
You can get an advice by running
> sudo nginx -t

Best regards,

**Comment by F1SMV** [comment]
thank , but in what directory i write this cmd ? i try in  $ sudo nginx -t ,  dont find anything, same in FAASGS directory, same /etc/nginx without success.. sorry

**Comment by takagiwa** [comment]
Thank you for trying. If you had seen nginx welcome page, usually you could run nginx command until uninstall.
In my environment (Raspberry Pi OS, apt package), nginx command is /usr/sbin/nginx .

Attached file is my configuration file ( /etc/nginx/sites-available/default ) . please rename default.txt to default

[default.txt](https://github.com/stdevPavelmc/FAASGS/files/5975100/default.txt)

**Comment by F1SMV** [comment]
Thank Takagiwa..
in first i try in /usr/sbin cmd sudo nginx -t and i had : nginx: [emerg] unexpected end of file, expecting "}" in /etc/nginx/sites-enabled/default:92
nginx: configuration file /etc/nginx/nginx.conf test failed
and after i switch default by your default setting...and...stay to find good setup ..and run predict. Many Thank for help

![capture raspi](https://user-images.githubusercontent.com/64283688/107845130-89abca80-6dd9-11eb-8070-a179505a5310.PNG)

**Comment by takagiwa** [comment]
Thank you for trying. I'm grad to hear that worked!

**Comment by stdevPavelmc** [comment]
> I think "index.php" not added or on later than "index.html" in "index" parameter somewere in nginx configuration file.

If you follows the install procedure mentioned here: https://github.com/stdevPavelmc/FAASGS#installation-steps you get the index.html of /var/www/html removed by this line in the makefile: https://github.com/stdevPavelmc/FAASGS/blob/0418823b73c1f1b941d8e221e760e18381757da7/Makefile#L18 

So there must be a issue with your install.

Please follow the upgrade procedure in the README.md (you will not lose any recording made)

**Comment by stdevPavelmc** [comment]
> Hello,
> 
> It looks there are syntax error on nginx configuration file.
> You can get an advice by running
> 
> > sudo nginx -t
> 
> Best regards,

FAASGS does not mess/deal with nginx config, it just uses the default site.

So if you are getting trouble please remove it and force a re-install of FAASGS like this **Inside the FAASGS folder**: (your recording are preserved)

```
apt remove nginx -y
make clean
make remove
make cconf
# configure your data in `/etc/sat_data/user.conf`
make install
make permanent
```

That will reset the setup and re-install the soft from scratch.

**Comment by stdevPavelmc** [comment]
> Thank Takagiwa..
> in first i try in /usr/sbin cmd sudo nginx -t and i had : nginx: [emerg] unexpected end of file, expecting "}" in /etc/nginx/sites-enabled/default:92
> nginx: configuration file /etc/nginx/nginx.conf test failed
> and after i switch default by your default setting...and...stay to find good setup ..and run predict. Many Thank for help
> 
> ![capture raspi](https://user-images.githubusercontent.com/64283688/107845130-89abca80-6dd9-11eb-8070-a179505a5310.PNG)

Just follow the above comment instructions and you will be fine.

**Comment by F1SMV** [comment]
sorry but my user.conf is ok, always "loc name" and soft dont run..and 
$ sudo ./sat.sh
Updating data and scheduling new sats
Fresh data fetched, updating the good file
Filtering the sats we need and set simple names
./sat.sh: line 301: /usr/bin/jq: No such file or directory
./sat.sh: line 48: /usr/bin/jq: No such file or directory

i had no modifications in sat.sh

bonne journée

**Comment by stdevPavelmc** [comment]
> ./sat.sh: line 301: /usr/bin/jq: No such file or directory
> ./sat.sh: line 48: /usr/bin/jq: No such file or directory

Please tell me more about your setup (Platfomr: Pc//Orange Pi/ Odroid/Raspberry Pi/etc, OS: RaspiOS, Armbian, etc...) as this tool must came by default on common distros.

Usually on debian like systems you fix that with a simple `sudo apt install jq -y`

**Comment by F1SMV** [comment]
no prob..raspi 4, Linux raspberrypi 5.10.11-v7l+ #1399 SMP Thu Jan 28 12:09:48 GMT 2021 armv7l GNU/Linux Debian
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
-------

i test apt install jq -y ..seems ok and after 
sudo apt sudo ./sat.sh
Updating data and scheduling new sats
Fresh data fetched, updating the good file
Filtering the sats we need and set simple names
=> ISS (ZARYA)/ISS
./sat.sh: line 89: [: file: integer expression expected
=> RADFXSAT (FOX-1B)/AO91-FOX-1B
./sat.sh: line 89: [: file: integer expression expected
=> FOX-1CLIFF (AO-95)/AO95-FOX-1CLIFF
./sat.sh: line 89: [: file: integer expression expected
=> FOX-1D (AO-92)/AO92-FOX-1D
./sat.sh: line 89: [: file: integer expression expected
=> DIWATA-2B/PO101-DIWATA-2B
./sat.sh: line 89: [: file: integer expression expected
=> NOAA 19/NOAA19
./sat.sh: line 89: [: file: integer expression expected
=> NOAA 18/NOAA18
./sat.sh: line 89: [: file: integer expression expected
=> NOAA 15/NOAA15
./sat.sh: line 89: [: file: integer expression expected

hope will help you

**Comment by stdevPavelmc** [comment]
Ok, I will set jq as a required dependency (it it's installed already then no problem) I just filled a BUG report, Will fix ASAP.

Now the error you are seeing is due to predict soft lack of configuration, yo need the follow this steps:

```
sudo mkdir /root/.predict
sudo touch /root/.predict/predict.qth
```

Then try to run `sats.sh` to see if it works now.

**Comment by F1SMV** [comment]
gm
seems run ..stay to find why my call isnt in window :) , run today i wait next passes to see in action. little problem , i have lna specially for noaa, i am be afraid that cut the amplification after 138 mhz..will see
![Capturesat](https://user-images.githubusercontent.com/64283688/108026747-3af46f80-7029-11eb-989f-02621ee7729e.PNG)

**Comment by F1SMV** [comment]
nothing...
![Capturesat](https://user-images.githubusercontent.com/64283688/108030438-87db4480-702f-11eb-8b33-1ba0cf9201dd.PNG)

**Comment by stdevPavelmc** [comment]
> gm
> seems run ..stay to find why my call isnt in window :) , run today i wait next passes to see in action. little problem , i have lna specially for noaa, i am be afraid that cut the amplification after 138 mhz..will see

The `make install` step must had take care of that, in fact it replaced the other data but no the LOC_NAME, can you show here the first lines of your `/etc/sat_data/user.conf` file, specially the ones shown below:

LOC=FL11bj
LOC_NAME="Camaguey"
LOC_COUNTRY="Cuba"

**Comment by stdevPavelmc** [comment]
>  i have lna specially for noaa, i am be afraid that cut the amplification after 138 mhz..will see

I hope not, even the well tuned LNA for VHF has a -3dB of at least a couple of Mhz, usually you get useful amplification on about 5 to 12 Mhz around the target passband.

Time will tell.

**Comment by stdevPavelmc** [comment]
> nothing...

Nothing as no image on the right?

**Comment by F1SMV** [comment]
CALL=F1SMV
NAME=Eric
LOC=JN23WB
LOC_NAME="La Seyne Sur Mer"
LOC_COUNTRY="France"

i switch JN23WB by JN23wb..
--------------------------------------

no..no pic..i try again today

**Comment by takagiwa** [comment]
Hello,

You may get something by running below on terminal when update schedule or trying to receive something.
sudo tail -f /var/mail/mail
You can stop by Ctrl+c .

Best regards,
Takagiwa

**Comment by takagiwa** [comment]
Hello 

I have checked nginx and php issue on Raspberry Pi.

When I access to root (http://sdr/ with my setup), I got "Welcome to nginx!"
When I open  http://sdr/index.php , browser tried to download index.php.
When I run "curl http://localhost/index.php", I could get php file (php not ran).

When I open http://sdr/sat/ , I got 403.
http://sdr/sat/20210220-192932_NOAA18/ also returns 403.
http://sdr/sat/20210220-192932_NOAA18/index.php was same as /index.php
But I can see http://sdr/sat/20210220-192932_NOAA18/NOAA18C.jpg .

My /var/www/html is below.

> root@sdr:/var/www/html# ls -alF
> total 124
> drwxrwx---  4 www-data www-data  4096 Feb 20 15:39 ./
> drwxrwx---  3 www-data www-data  4096 Feb 20 15:24 ../
> drwxrwx---  2 www-data www-data  4096 Feb 20 15:39 img/
> -rw-rw----  1 www-data www-data   612 Feb 20 15:24 index.nginx-debian.html
> -rw-rw----  1 www-data www-data  7364 Feb 20 15:39 index.php
> -rw-rw----  1 www-data www-data 86927 Feb 20 15:39 jquery.min.js
> drwxrwx--- 25 www-data www-data  4096 Feb 21 12:56 sat/
> -rw-rw----  1 www-data www-data  4530 Feb 20 15:39 style.css

My understanding are ...
* php file not handled by nginx automatically just after install both.
* index.php no added to index automatically.
* default index file was index.nginx-debian.html .

So I think Raspberry Pi OS user have to modify nginx configuration file like my attached file above.

Best regards,

**Comment by takagiwa** [comment]
Hello

Because of there are no index.html on Raspberry Pi OS, removing that file on Makefile needs check file before do that.
The command "sudo find /var/www/html/index.html -delete" maybe good for other OS. Ignoring error required on Raspberry Pi OS. 


Best regards,

**Comment by stdevPavelmc** [comment]
Hum...

Good insight, I have secured a Rasberry Pi to test it on it, I have to pick up a Rasberry Pi 3B+ later today.

Remember my actual dev platform is Armbian over a Orange Pi Prime board, I will do the testing over RaspiOS now and will incorporate fixes for all that issues.

Cheers.

**Comment by takagiwa** [comment]
Yes I know.
My post is an information for other Raspberry Pi users.

Best regards,

**Comment by stdevPavelmc** [comment]
> CALL=F1SMV
> NAME=Eric
> LOC=JN23WB
> LOC_NAME="La Seyne Sur Mer"
> LOC_COUNTRY="France"
> ## i switch JN23WB by JN23wb..
> 
> no..no pic..i try again today

This issue was related to city/country having spaces, and was addressed today with a patch, please update, follow the README.md instructions for that

**Comment by stdevPavelmc** [comment]
> Hello
> 
> I have checked nginx and php issue on Raspberry Pi.
> 
> When I access to root (http://sdr/ with my setup), I got "Welcome to nginx!"
> When I open http://sdr/index.php , browser tried to download index.php.
> When I run "curl http://localhost/index.php", I could get php file (php not ran).
> 
> When I open http://sdr/sat/ , I got 403.
> http://sdr/sat/20210220-192932_NOAA18/ also returns 403.
> http://sdr/sat/20210220-192932_NOAA18/index.php was same as /index.php
> But I can see http://sdr/sat/20210220-192932_NOAA18/NOAA18C.jpg .
> 
> My /var/www/html is below.
> 
> > root@sdr:/var/www/html# ls -alF
> > total 124
> > drwxrwx---  4 www-data www-data  4096 Feb 20 15:39 ./
> > drwxrwx---  3 www-data www-data  4096 Feb 20 15:24 ../
> > drwxrwx---  2 www-data www-data  4096 Feb 20 15:39 img/
> > -rw-rw----  1 www-data www-data   612 Feb 20 15:24 index.nginx-debian.html
> > -rw-rw----  1 www-data www-data  7364 Feb 20 15:39 index.php
> > -rw-rw----  1 www-data www-data 86927 Feb 20 15:39 jquery.min.js
> > drwxrwx--- 25 www-data www-data  4096 Feb 21 12:56 sat/
> > -rw-rw----  1 www-data www-data  4530 Feb 20 15:39 style.css
> 
> My understanding are ...
> 
>     * php file not handled by nginx automatically just after install both.
> 
>     * index.php no added to index automatically.
> 
>     * default index file was index.nginx-debian.html .
> 
> 
> So I think Raspberry Pi OS user have to modify nginx configuration file like my attached file above.
> 
> Best regards,

Addressed via #26, will put a comment in the readme with a link to a tutorial from the raspi foundation about enabling php on nginx...
